app/controllers/documents_v2_controller.rb

```diff
@@ -36,9 +35,14 @@ def save
 
     return unless require_valid_read_write_access_key
 
-    @document.form_content = request.raw_post
+    if params[:reset].present?
+      @document.content = @document.original_content
+    else
+      @document.form_content = request.raw_post
+    end
+
```

Comment:
As we discussed on zoom. The original_content shouldn't be used because the authored document might have changed. And in that case reset to original should show the latest version of the authored document.